crypto/ec/ec_check.c

```diff
@@ -22,6 +22,13 @@ int EC_GROUP_check_named_curve(const EC_GROUP *group, int nist_only)
 
 int EC_GROUP_check(const EC_GROUP *group, BN_CTX *ctx)
 {
+#ifdef FIPS_MODE
+    /*
+    * ECC domain parameter validation.
+    * See SP800-56A R3 5.5.2 "Assurances of Domain-Parameter Validity" Part 1b.
+    */
+    return EC_GROUP_check_named_curve(group, 1) >= 0 ? 1 : 0;
```

Comment:
The ` ? 1 : 0` isn't required here.  C defines conditional operators as return 0 or 1.

Reply:
If you remove the ```?1:0``` then what is the conditional operator?
How do you know, for example, that it's not returning -1?

Reply:
The conditional operator would be `>=`.
The C standard defines them as return 0 or 1.

Reply:
oh right, sorry.